pkg/controller/registry/resolver/evolver.go

```diff
@@ -112,8 +112,15 @@ func (e *NamespaceGenerationEvolver) queryForRequiredAPIs() error {
 		}
 		e.gen.MarkAPIChecked(*api)
 
+		// identify the initialSource
+		initialSource := CatalogKey{}
+		for _, operator := range e.gen.MissingAPIs()[*api] {
+			initialSource = operator.SourceInfo().Catalog
```

Comment:
Correct me if I'm wrong, but doesn't this break down when there exists two or more concurrent `Subscriptions` that require the same API, but target different sources that contain operators providing that API?

Reply:
You are correct that the current approach can introduce some randomness when the following conditions are met:

* `Operator A` defines dependencies on APIs provided by`Operator B` and `Operator C`
* `Operator B` and `Operator C` come from different `CatalogSources`
* Both `Operator B` and `Operator C` define dependencies on the same API provided by `Operator D`.

In this situation - the existing solution does not guarantee that a specific `CatalogSource` will be prioritized consistently.

This is a known issue with the short-term solution and is expected to be addressed in a future solution. We are comfortable with supporting the existing approach due to the fact that this "bug" only hits a small number of use cases.